### ISSUE TITLE
Create Questions table

### ISSUE THREAD

**Issue by drulabs** [UdacityAndroidDevScholarship/quiz-app#48]
We are also providing create quiz functionality, for this, we need to **store information locally** till the quiz creator enters all the information and submits the quiz.

For example: for an SQLite Quiz, title, the lesson can be stored in shared preferences. But for other stuff like questions (30 Max) and options for each question (2-6 options) we need to store in local SQLite database and delete it only after the quiz is successfully submitted to the backend (firebase).

1 quiz -> N question -> every question can have 2 to 6 options, 0 options for subjective questions.

Create an SQLite table schema for storing questions and options and relate them. Make sure the tables are normalized. You may use referential integrity (foreign keys).

Please discuss the schema using markdown syntax in comments. For implementation, a separate ticket will be created.

**Comment by drulabs** [comment]
@pramodbharti, please go ahead. Please note that we need the schema before implementation.

**Comment by vivekpanchal** [comment]
would  work on this

**Comment by drulabs** [comment]
@vivekpanchal sure. Please work on the schema as described in the issue before writing java code.

**Comment by huzefaraja** [comment]
I would like to work on this.

Forgive me if this is a stupid question, but the "schema using markdown syntax in comments", is that supposed to be here in the comments?

**Comment by drulabs** [comment]
@huzefaraja meaning. Please discuss your approach, the tables or tables you are going to use, any connection etc. here in issue comments.

**Comment by drulabs** [comment]
@huzefaraja the schema looks great, please start working on it. You need to create Contracts for these, tables and a `private static final String CREATE_<table_name>_TABLE` variable in `DBHelper.java`. Execute these in the onCreate() method. Leave onUpgrade() blank for now. Please make sure you follow the SQLite lesson of the course.

Please make `Options.Remark` column nullable rest of the columns will be not null. Also add a field called `quiz_identifier` (String), to identify questions from different quizzes.

Please do not add anything else (no onUpgrade(), no delete table statement, no adapters), just the contract files and DBHelper.java

**Others**: If you agree with this schema, please start working on it. Do give credit to @huzefaraja's in the comments of contract and DBHelper classes. If not please discuss the schema here.

**Comment by pramodbharti** [comment]
@drulabs . Here is my schema. Please review this.
                                         TABLE NAME : TBL_QUIZ_DETAILS
Fields | Data Type | Constraints | Remarks
----------------- | ----------------- | ---------------- | -----------------
QUIZ_ID | INTEGER | PRIMARY KEY + AUTO_INCREMENT | 
CREATOR_ID | INTEGER | NOT NULL | 
CREATOR_NAME | TEXT | NOT NULL | 
QUIZ_TITLE | TEXT | NOT NULL | 
LESSON | TEXT | NOT NULL | 
DESC | TEXT | | 
LAST_MODIFIED | CURRENT_TIMESTAMP | NOT NULL | 
DEADLINE | DATE | | If any deadline
MAX_MARKS | INTEGER | NOT NULL | 
TOTAL_QUES | INTEGER | NOT NULL | For storing total no. of questions
BG_IMG | TEXT | | For storing bg image's path if any
DIFFCULTY | INTEGER | NOT NULL | For storing difficulty level EASY(0), MEDIUM(1), HARD(2). We can use predefined constant here.

 
                                         TABLE NAME : TBL_QUIZ_QUESTIONS
Fields | Data Type | Constraints | Remarks
----------------- | ----------------- | ---------------- | -----------------
QUE_ID | INTEGER | PRIMARY KEY, AUTO_INCREMENT | 
QUIZ_ID | INTEGER | FOREIGN KEY, NOT NULL | Foreign key referencing TBL_QUIZ_DETAILS QUIZ_ID
QUE_DESC | TEXT | NOT NULL | 
QUE_IMG | TEXT |  | For storing image's path if any.
QUE_TYPE | INTEGER | NOT NULL | For storing type of question, SUBJECTIVE(0), SINGLE_CHOICE(1), MULTIPLE_CHOICE(2). We can use predefined constant here.
MARKS | INTEGER | NOT NULL | 

                                         TABLE NAME : TBL_QUE_OPTIONS
Fields | Data Type | Constraints | Remarks
----------------- | ----------------- | ---------------- | -----------------
OPT_ID | INTEGER | PRIMARY KEY, AUTO_INCREMENT | 
QUE_ID | INTEGER | FOREIGN KEY, NOT NULL | Foreign key referencing TBL_QUIZ_QUESTIONS QUE_ID
OPT_DESC | TEXT | NOT NULL | Text or image path if any 
IS_CORRECT | BOOLEAN |  | 
OPT_REMARKS | TEXT | | Remarks if any

**Comment by pramodbharti** [comment]
Hi @drulabs  currently I am creating PR with two tables i.e. `TBL_QUESTIONS` and `TBL_OPTIONS` as described by you but I think we need another table `QUIZ_DETAILS`. We can discuss later.

**Comment by drulabs** [comment]
@pramodbharti we don't need a quiz details table as mentioned in the Issue `Create an SQLite table schema for storing questions and options and relate them`. Before implementation, you should have discussed the schema in comments as mentioned in comments of this issue. Here are some feedbacks:
- Instead of short abbreviations please use full names. e.g. QUESTION_TYPE instead of QUE_TYPE, TABLE_QUIZ instead of TBL_QUIZ
- COLUMN names are not as per SQLite lesson
- As clear from JSON that QUESTION_TYPE is a string, not an integer
- How to handle multiple images in QUE_IMG section?
- In the PR you've raised, I can see there is no relation between tables as mentioned in the schema (I will comment other code related stuff on the PR).

Please follow the JSON as it is, do not assume any other data type.

**Comment by pramodbharti** [comment]
@drulabs  I have made changes in schema. I think we need third table (`files`) for storing images. Please have a look.

**Comment by kesari-vamshi** [comment]
can you please give the node of resources  of firebase sample model
